src/app/shared/form/builder/form-builder.service.ts

```diff
@@ -116,22 +118,51 @@ export class FormBuilderService extends DynamicFormService {
     return { $event, context, control: control, group: group, model: model, type };
   }
 
-  getTypeBindModel() {
-    return this.typeBindModel.getValue();
+  /**
+   * Get the field model using the given field id. This is extracted from the array of models 'typeBindModels'.
+   * @param id The field id to find the corresponding model.
+   * @returns The model for the given field id.
+   */
+  getTypeBindModel(id: string) {
+    // Make sure to have the '_' separator since the keys of typeBindModels are using this separator.
+    id = this.formatField(id);
+    return this.typeBindModels[id]?.getValue();
   }
 
-  getTypeBindModelUpdates(): Observable<any> {
-    return this.typeBindModel.pipe(
-      distinctUntilChanged(),
-      switchMap((bindModel: any) => {
-        return (bindModel.type === 'CHECKBOX_GROUP' ? bindModel.valueUpdates : bindModel.valueChanges);
-      }),
-      distinctUntilChanged(),
-    );
+  /**
+   * Get the field model value change event as an observable using the given field id.
+   * This is extracted from the array of models 'typeBindModels'.
+   * @param id The field id to find the corresponding model.
+   * @returns An observable of the value change event from the model found using the given field id.
+   * Can be Observable<null> if no model found for the given id.
+   */
+  getTypeBindModelUpdates(id: string): Observable<any> {
+    let model = this.typeBindModels[id];
+    return hasValue(model)
+      ? model.pipe(
+        distinctUntilChanged(),
+        switchMap((bindModel: any) => {
+          // Check the type of the type-bind field.
+          return (bindModel.type === 'CHECKBOX_GROUP' ? bindModel.valueUpdates : bindModel.valueChanges);
+        }),
+        distinctUntilChanged(),
+      )
```

Comment:
small indentation problem to have readable code. Add 2 spaces for lines into the `pipe`; indent last ')'